composition-js/src/merging/merge.ts

```diff
@@ -161,25 +163,15 @@ function isGraphQLBuiltInDirective(def: DirectiveDefinition): boolean {
   return !!def.schema().builtInDirective(def.name);
 }
 
-function isMergedDirective(definition: DirectiveDefinition | Directive): boolean {
-  // If it's a directive application, then we skip it unless it's a graphQL built-in
-  // (even if the definition itself allows executable locations, this particular
-  // application is an type-system element and we don't want to merge it).
-  if (definition instanceof Directive) {
-    // We have special code in `Merger.prepareSupergraph` to include the _definition_ of @tag in the supergraph
-    // because we want to make sure it's included as a "core feature", and so we want this method
-    // to return `false` for `@tag` definition, but we *do* want to merge `@tag` applications so we
-    // special case it.
-    // Note that this is a temporary solution: a more principled way to have directive propagated
-    // is coming and will remove the hard-coding.
-    return definition.name === 'tag' || isGraphQLBuiltInDirective(definition.definition!);
-  } else if (isGraphQLBuiltInDirective(definition)) {
-    // We never "merge" graphQL built-in definitions, since they are built-in and
-    // don't need to be defined.
-    return false;
-  }
-  return definition.locations.some(loc => executableDirectiveLocations.includes(loc));
+type MergedDirectiveInfo = {
+  specInSupergraph: FeatureDefinition,
+  definitionInSubgraph: (subgraph: Subgraph) => DirectiveDefinition,
 }
+
+const MERGED_FEDERATION_DIRECTIVES: MergedDirectiveInfo[] = [
```

Comment:
If someone wonders why this patch is kind of over-generic despite only have 1 merged directive, that's because `@inaccessible` will be add to this list shortly.